l10n_it_split_payment/models/account_move.py

```diff
@@ -40,3 +41,36 @@ def _compute_amount(self):
             else:
                 move.amount_sp = 0.0
         return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        if self.env.context.get("skip_split_payment_computation"):
+            return res
+        for move in self:
+            if move.split_payment:
+                split_payment_total = 0
+                for line in move.line_ids:
+                    if line.display_type == "tax" and not line.is_split_payment:
+                        write_off_line_vals = line._build_writeoff_line()
+                        split_payment_total += write_off_line_vals["debit"]
+                        line_sp = fields.first(
```

Comment:
Per ogni riga cerca tra tutte le righe della `move` e troverà sempre la stessa `line_sp`, non sarebbe meglio salvarla? Così almeno evitiamo di rifare la ricerca e il `filtered` che tra l'altro ora è anche su un campo calcolato.

Se poi riuscissimo anche a fare la scrittura fuori dal loop non sarebbe male.

In pratica, ora che siamo nella `move` si potrebbe riscrivere un po' questo loop per evitare di rifare le stesse operazioni per ogni riga: ad esempio ottenere in un colpo solo l'importo da mettere nel `writeoff` da tutte le righe che non sono `is_split_payment`, visto che `_build_writeoff_line` prende quasi tutti i dati dalla `move`; cosa ne pensi?

Reply:
Ho tirato fuori `line_sp` ma non mi è chiara l'altra proposta